interfaces/acados_matlab_octave/detect_dims_sim.m

```diff
@@ -52,12 +52,18 @@
         model.dim_np = 0;
     end
 
+    if isfield(model, 'sym_xdot')
+        if numel(model.sym_xdot) ~= model.dim_nx
+            warning('sym_xdot is not of shape nx');
```

Comment:
error instead of warning here?

Reply:
`xdot` is not needed for explicit integrator.
So, I went for warning.